### PR TITLE
added json output to whisper-diff.py script

### PR DESCRIPTION
**Pull request by diuis**
Added json output to the whisper-diff.py script.
The json output could be enabled with the --json parameter (the same that I found in the whisper-fetch.py script; in my PR I use the json python library (import json).
I created also a main() function (as in the whisper-fill.py script) to enable the import of whisper-diff functions in another scripts.

### CODE REVIEW

File: bin/whisper-diff.py [hunk 1]
```diff
@@ -63,9 +66,25 @@ def print_summary(diffs,pretty=True,headers=True):
...
-    
-archive_diffs = whisper.diff(path_a,path_b,ignore_empty=options.ignore_empty,until_time=until_time)
-if options.summary:
-  print_summary(archive_diffs,pretty=(not options.columns),headers=(not options.no_headers))
-else:
-  print_diffs(archive_diffs,pretty=(not options.columns),headers=(not options.no_headers))
\ No newline at end of file
+
+def print_summary_json(diffs,path_a,path_b):
+  print json.dumps({'path_a': path_a, 'path_b': path_b, 'archives': [{'archive': archive, 'total': total, 'points': points.__len__()} for archive, points, total in diffs]}, sort_keys=True, indent=2, separators=(',', ' : '))
```
Comment (obfuscurity) [inline bin/whisper-diff.py]: These lines are way too long. We're not going to be overly pedantic about PEP conventions here, but we should try to limit them to a reasonable length. See https://www.python.org/dev/peps/pep-0008/#maximum-line-length for some guidelines.

**Comment by diuis** [comment]
Thanks for the suggestion @obfuscurity .
I think now the code is more readable: take a look and let me know.

File: bin/whisper-diff.py (hunk 1 above)
Comment (diuis) [inline bin/whisper-diff.py]: Hi @obfuscurity , did you take a look at the new implementation?

**Comment by SEJeff** [comment]
The `if __name__ == '__main__':` changes are good and pythonic but Python modules to be imported can't have a dash "-" in the name. You could never do `import whisper-diff`.

**Comment by diuis** [comment]
Hi @SEJeff ,
I'm aware that I can't import a Python module with a dash in the filename.
I'd like to use a solution like this:

```
---- foo_proxy.py ----
 tmp = __import__('foo-bar')
 globals().update(vars(tmp))

 ---- main.py ----
 from foo_proxy import * 
```

http://stackoverflow.com/questions/8350853/how-to-import-python-module-when-module-name-has-a-dash-or-hyphen-in-it

What do you think about it?

File: bin/whisper-diff.py (hunk 1 above)
Comment (diuis) [inline bin/whisper-diff.py]: Hi @obfuscurity do you need some other changes?

Comment (diuis) [inline bin/whisper-diff.py]: Thanks @obfuscurity, could you accept the PR?

**Comment by SEJeff** [comment]
The globals trick is a clever hack, but I would really not like to see that kind of code upstream. If you wanted to use it elsewhere totally cool with that.

@obfuscurity: merge at will. The change was good, but the comment on why he did it wasn't.

**Pull request merged**